my-todo/src/components/TodoForm.js

```diff
@@ -0,0 +1,102 @@
+
+import React, { useState } from "react";
+import {makeStyles} from '@material-ui/core';
+import {Button} from '@material-ui/core';
+//import AddIcon from '@material-ui/icons/Add';
+//import DeleteIcon from '@material-ui/icons/Delete'
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import TextField from '@material-ui/core/TextField';
+import Grid from '@material-ui/core/Grid';
+import MomentUtils from '@date-io/moment';
+import {
+  DateTimePicker,
+  MuiPickersUtilsProvider
+} from "@material-ui/pickers";
+import * as moment  from 'moment'
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    flexGrow: 1
+  },
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap'
+  },
+  addItemTextField:{
+    marginLeft: theme.spacing(1),
+    marginRight: theme.spacing(1),
+  },
+  dense: {
+    marginTop: theme.spacing(2)
+  },
+  menu: {
+    width: 200
+  },
+  menuButton: {
+    marginRight: theme.spacing(2)
+  },
+  title: {
+    flexGrow: 1
+  }
+}));
+
+export const TodoForm = ({dispatch}) => {
+
+    const classes = useStyles();
+
+    const [item, setItem] = useState("");
+
+    const [selectedDate, handleDateChange] = useState(new Date());
+
+
+    const handleChanges = event => {
+        setItem(event.target.value);
+        handleDateChange(event.target.value)
+    };
+
+    const submitForm = event => {
+        event.preventDefault();
+        dispatch({
+            type: "ADD_ITEM",
+            payload: item || item.title
+        });
+        dispatch({
+            type: "ADD_DATE",
+            payload: selectedDate || selectedDate.title
+        });
+        setItem("");
+        handleDateChange("")
```

Comment:
moment library errors out with an empty string as a value.  You need it to be null